### PR TITLE
MAT8599 test case validation status and message

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR Template

Jira Ticket: [MAT-MAT8599](https://jira.cms.gov/browse/MAT-MAT8599)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/measure/TestCase.java [hunk 1]
```diff
@@ -65,6 +68,9 @@ public class TestCase implements Serializable, Cloneable {
 
   @Valid private List<TestCaseGroupPopulation> groupPopulations;
 
+  private String testCaseValidationStatus;
+  private String testCaseValidationMessage;
```
Comment (adongare) [inline src/main/java/gov/cms/madie/models/measure/TestCase.java]: @nmorasb @RohitKandimalla- are we going to capture just the validation message or the location in the json as well? maybe depends on usecase? if we want to provide scrollto feature upon clicking the error, we might need location of the error. and there could be multiple messages

Comment (adongare) [reply src/main/java/gov/cms/madie/models/measure/TestCase.java]: okay, we already have one-
`@Transient private HapiOperationOutcome hapiOperationOutcome;`
Should we just repurpose it? maybe we can rename it or leave it the way it is.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/measure/TestCase.java]: @adongare @nmorasb The card specifies to have a validation status field of enum type and a validation message. I don't think we can repurpose it at this point, because hapiOperationOutcome is still used in both frontend and backend. And I don't know what value to map into the status enum.

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/models/measure/TestCase.java]: We can use hapiOperationOutcome as the validation message, and then validationStatus should be an enum.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/measure/TestCase.java]: @RohitKandimalla Ok thanks, I'll remove testCaseValidationMessage then.

Comment (adongare) [reply src/main/java/gov/cms/madie/models/measure/TestCase.java]: Are we going to store the validation messages?  if yes- hapioperationoutcoe is transient. we will have to remove the transient annotation so that it can be persisted. ~~and I think its going to be a list?~~

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/measure/TestCase.java]: @adongare Yeah I think we are storing the validation message, removed transient annotation.

**Review by adongare — APPROVED** [review]
